src/fileformat/types/resource_table/bitmap_image.cpp

```diff
@@ -262,22 +207,23 @@ bool BitmapImage::parseDib1Data(const ResourceIcon &icon, const struct BitmapInf
 
 		for (std::size_t j = 0; j < nColumns / 8; j++)
 		{
-			for (auto bit = 7; bit >= 0; bit--)
+			for (std::size_t i = 0; i < 8; i++)
 			{
-				auto idx = (bytes[offset] >> bit) & 0x01;
-				row.push_back(palette[idx]);
+				auto index = !!(bytes[offset] & (0x01 << (7 - i)));
```

Comment:
Personally, I find code that uses `!!` unnecessarily hard to read. Why not just using a `bool` or explicit `1`/`0`? Generally, I suggest preferring clarity over brevity.